Sources/XCTest/XCTestObservation.swift

```diff
@@ -77,3 +77,13 @@ public extension XCTestObservation {
     func testSuiteDidFinish(_ testSuite: XCTestSuite) {}
     func testBundleDidFinish(_ testBundle: NSBundle) {}
 }
+
+/// Expanded version of `XCTestObservation` used internally to respond to
+/// additional events not publicly exposed.
+internal protocol _XCTestObservation: XCTestObservation {
```

Comment:
At the risk of posting another nit-pick, I had to re-read this two or three times before I realized `_XCTestObservation` was not the same as `XCTestObservation`. How about `XCTestInternalObservation`, to emphasize the difference and alleviate your concerns of an underscore prefix?